gio/src/action_entry.rs

```diff
@@ -0,0 +1,120 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{ActionMap, SimpleAction};
+use glib::{IsA, Variant};
+
+#[doc(alias = "GActionEntry")]
+pub struct ActionEntry<O>
+where
+    O: IsA<ActionMap>,
+{
+    name: String,
+    parameter_type: Option<String>,
+    state: Option<String>,
+    pub(crate) activate: Option<Box<dyn Fn(&O, &SimpleAction, Option<&Variant>) + 'static>>,
+    pub(crate) change_state: Option<Box<dyn Fn(&O, &SimpleAction, Option<&Variant>) + 'static>>,
+}
+
+impl<O> ActionEntry<O>
+where
+    O: IsA<ActionMap>,
+{
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn parameter_type(&self) -> Option<&str> {
+        self.parameter_type.as_deref()
+    }
+
+    pub fn state(&self) -> Option<&str> {
+        self.state.as_deref()
+    }
+
+    pub fn builder(name: &str) -> ActionEntryBuilder<O> {
+        ActionEntryBuilder::new(name)
+    }
+}
+
+impl<O> std::fmt::Debug for ActionEntry<O>
+where
+    O: IsA<ActionMap>,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ActionEntry")
+            .field("name", &self.name())
+            .field("parameter_type", &self.parameter_type())
+            .field("state", &self.state())
+            .finish()
+    }
+}
+
+#[derive(Debug)]
+pub struct ActionEntryBuilder<O>(ActionEntry<O>)
+where
+    O: IsA<ActionMap>;
```

Comment:
All these `where` clauses are not actually needed, are they?

Reply:
From my tests, either we had to add those where clauses or you will have to specify the type when constructing the ActionEntry itself.

Reply:
Ah I would've expected the compiler to figure that out when the action entries are added to some actionmap